### DISCUSSION TITLE
Ignore {{ }} Expressions in included files

### DISCUSSION THREAD

**Discussion by RhodriMMD** [helmfile/helmfile#1040] [Questions] [answered]
helmfile version 0.156.0

I am currently setting up a helmfile that sets up a grafana stack and a few other things.
I have hit a snag when trying to add dashboards however as some of them also use {{ as expressions such as https://grafana.com/grafana/dashboards/11001-cert-manager/

I have tried adding the JSON to the configmap manually and escaping the expression however this also then breaks the dashboard.
Though i have not attempted {{ printf }} yet, but currently looking for a way that does not include editing the file.
I have also tried loading the file using .Files or readFile and still get the below error

parse error at (K8SToolbox-Dashboards/templates/dashbaords/cert-manager.json:135): function "exported_namespace" not defined

Is there something I missed, as found a few similar issues but they are usually not a full JSON file like this?

**Comment by yxxhero** [comment]
@RhodriMMD please post a example.

**Comment by RhodriMMD** [comment]
Here is what i have so far, the .json part has changed a few times in my testing.
``
apiVersion: v1
kind: ConfigMap
metadata:
  namespace: monitoring
  name: {{ .Release.Name }}-kube-prometheus-cert-manager
  labels:
      app: kube-prometheus-stack-grafana
      app.kubernetes.io/part-of: kube-prometheus-stack
      grafana_dashboard: '1'
data:
  cert-manager-dash.json: |
    {{ readFile "dashboards/cert-manager.json" | nindent 4 }}
``

and the json file it is importing will be from https://grafana.com/grafana/dashboards/11001-cert-manager/ (wont dump it here as its tad large)

And finally the error i get is below where "exported_namespace" is part of the file.

`helmfile template --environment=test -f helmfile.d/10-dashboards.yaml
WARNING: environments and releases cannot be defined within the same YAML part. Use --- to extract the environments into a dedicated part
Building dependency release=dashboards, chart=../charts/dashboards
Templating release=dashboards, chart=../charts/dashboards
in helmfile.d/10-dashboards.yaml: command "/usr/sbin/helm" exited with non-zero status:

PATH:
  /usr/sbin/helm

ARGS:
  0: helm (4 bytes)
  1: template (8 bytes)
  2: dashboards (10 bytes)
  3: ../charts/dashboards (20 bytes)
  4: --version (9 bytes)
  5: 0.0.1 (5 bytes)

ERROR:
  exit status 1

EXIT STATUS
  1

STDERR:
  Error: parse error at (K8SToolbox-Dashboards/templates/dashbaords/cert-manager.json:135): function "exported_namespace" not defined
  Use --debug flag to render out invalid YAML

COMBINED OUTPUT:
  Error: parse error at (K8SToolbox-Dashboards/templates/dashbaords/cert-manager.json:135): function "exported_namespace" not defined
  Use --debug flag to render out invalid YAML`

**Comment by RhodriMMD** [comment, marked as answer]
Ok solved my issue and was my fault lol
I had the "dashboards" folder in templates
moving that out and using 
{{- (.Files.Glob "dashboards/**.json").AsConfig | nindent 2 }}
fixed my issue

I forgot the "root" was not where the file was but root of the chart and also forgot anything in templates gets rendered..
